packages/lib-classifier/src/plugins/tasks/survey/components/SurveyTask.spec.js

```diff
@@ -1,51 +1,566 @@
+        const { getAllByRole, getByRole, queryByRole } = render(<NoFiltersStory />)
+        // tabbing to the first choice (Aardvark), arrowing up to the last choice (Nothing here), and pressing Enter to open the choice (Nothing here)
+        await user.keyboard('[Tab]')
+        await user.keyboard('[ArrowUp]')
+        await user.keyboard('[Enter]')
+        let choiceHeading = getByRole('heading', { name: 'Nothing here' })
+        expect(choiceHeading).to.be.ok()
+        // tabbing to the "Identify" button
+        await user.keyboard('[Tab][Tab][Tab]')
+        const identifyButton = getByRole('button', { name: 'SurveyTask.Choice.identify' })
+        expect(identifyButton).to.equal(document.activeElement)
+        // pressing Enter to identify and close the choice (Nothing here)
+        await user.keyboard('[Enter]')
+        // confirm choice (Nothing here) heading, and therefore choice, is not shown
+        choiceHeading = queryByRole('heading', { name: 'Fire' })
+        expect(choiceHeading).to.be.null()
+        // confirm choices are shown
+        const choiceButtons = getAllByRole('menuitemcheckbox')
+        expect(choiceButtons.length).to.equal(6)
+        // confirm the identified choice (Nothing here) is the active choice
+        const nothingHereChoiceButton = getByRole('menuitemcheckbox', { name: 'Nothing here' })
+        expect(nothingHereChoiceButton).to.equal(document.activeElement)
+      })
 
-    it('should not render a Chooser component', function () {
-      expect(wrapper.find(Chooser)).to.have.lengthOf(0)
+      it('should disable the Identify button until required questions are answered', async function () {
+        const user = userEvent.setup({ delay: null })
+        const { getByRole } = render(<NoFiltersStory />)
+        // tabbing to the first choice (Aardvark) and pressing Enter to open the choice (Aardvark)
+        await user.keyboard('[Tab][Enter]')
+        let identifyButton = getByRole('button', { name: 'SurveyTask.Choice.identify' })
+        // confirm the Identify button is disabled, pending required questions answered
+        expect(identifyButton.disabled).to.be.true()
+        // the required questions for Aardvark are "How many?" and "What behavior do you see?"
+        // the following answers "How many?" with "1" and "What behavior do you see?" with "Eating"
+        
+        // tabbing (x3) to the "How many?" question, selecting the "1" answer with space key, tabbing (x4) to the "What behavior do you see?" question, selecting the "Eating" answer with space key
+        await user.keyboard('[Tab][Tab][Tab][Space][Tab][Tab][Tab][Tab][Space]')
+        // confirm the Identify button is enabled, now that required questions are answered
+        identifyButton = getByRole('button', { name: 'SurveyTask.Choice.identify' })
+        // confirm the Identify button is enabled, now that required questions answered
+        expect(identifyButton.disabled).to.be.false()
+      })
     })
```

Comment:
I think I need to add a test that deletes an existing identified choice. Realized thanks to https://coveralls.io/builds/54306361.

Reply:
Added in [[b880fa0](https://github.com/zooniverse/front-end-monorepo/pull/3842/commits/b880fa00990b557f216c9933d400db6d842354cf)](https://github.com/zooniverse/front-end-monorepo/pull/3842/commits/b880fa00990b557f216c9933d400db6d842354cf)